src/StaticAllocator.zig

```diff
@@ -56,6 +57,6 @@ fn resize(self: *Self, buf: []u8, buf_align: u29, new_len: usize, len_align: u29
 }
 
 fn free(self: *Self, buf: []u8, buf_align: u29, ret_addr: usize) void {
-    assert(self.state == .deinit);
+    assert(self.state == .init or self.state == .deinit);
```

Comment:
Good catch!

We could then switch state to `deinit` on the first free.

Reply:
Done!